include/zephyr/drivers/mspi.h

```diff
@@ -0,0 +1,791 @@
+ * community to fill it up.
+ */
+enum mspi_bus_event_cb_mask {
+	MSPI_BUS_NO_CB              = 0,
+	MSPI_BUS_RESET_CB           = BIT(0),
+	MSPI_BUS_ERROR_CB           = BIT(1),
+	MSPI_BUS_XFER_COMPLETE_CB   = BIT(2),
+};
+
+/**
+ * @brief MSPI transfer modes
+ */
+enum eTransMode {
+	MSPI_PIO,
+	MSPI_DMA,
+};
+
+/**
+ * @brief MSPI transfer directions
+ */
+enum eTransDirection {
+	MSPI_RX,
+	MSPI_TX,
+};
+
+/**
+ * @brief MSPI controller device specific configuration mask
+ */
+enum mspi_dev_cfg_mask {
+	MSPI_DEVICE_CONFIG_NONE         = 0,
+	MSPI_DEVICE_CONFIG_CE_NUM       = BIT(0),
+	MSPI_DEVICE_CONFIG_FREQUENCY    = BIT(1),
+	MSPI_DEVICE_CONFIG_IO_MODE      = BIT(2),
+	MSPI_DEVICE_CONFIG_DATA_RATE    = BIT(3),
+	MSPI_DEVICE_CONFIG_CPP          = BIT(4),
+	MSPI_DEVICE_CONFIG_ENDIAN       = BIT(5),
+	MSPI_DEVICE_CONFIG_CE_POL       = BIT(6),
+	MSPI_DEVICE_CONFIG_DQS          = BIT(7),
+	MSPI_DEVICE_CONFIG_RX_DUMMY     = BIT(8),
+	MSPI_DEVICE_CONFIG_TX_DUMMY     = BIT(9),
+	MSPI_DEVICE_CONFIG_READ_INSTR   = BIT(10),
+	MSPI_DEVICE_CONFIG_WRITE_INSTR  = BIT(11),
+	MSPI_DEVICE_CONFIG_INSTR_LEN    = BIT(12),
+	MSPI_DEVICE_CONFIG_ADDR_LEN     = BIT(13),
+	MSPI_DEVICE_CONFIG_MEM_BOUND    = BIT(14),
+	MSPI_DEVICE_CONFIG_BREAK_TIME   = BIT(15),
+	MSPI_DEVICE_CONFIG_ALL          = BIT_MASK(16),
+};
+
+/**
+ * @brief MSPI XIP access permissions
+ */
+enum mspi_xip_permit {
+	MSPI_XIP_READ_WRITE     = 0,
+	MSPI_XIP_READ_ONLY      = 1,
+};
+
+/**
+ * @brief MSPI Configure API
+ * @defgroup mspi_configure_api MSPI Configure API
+ * @{
+ */
+
+/**
+ * @brief Stub for timing parameter
+ */
+enum mspi_timing_param {
+	MSPI_TIMING_PARAM_DUMMY
+};
+
+/**
+ * @brief Stub for struct timing_cfg
+ */
+struct mspi_timing_cfg {
+
+};
+
+/**
+ * @brief MSPI device ID
+ * The controller can identify its devices and determine whether the access is
+ * allowed in a multiple device scheme.
+ */
+struct mspi_dev_id {
+	/** @brief device gpio ce */
+	struct gpio_dt_spec     ce;
+	/** @brief device index on DT */
+	uint16_t                dev_idx;
+};
+
+/**
+ * @brief MSPI controller configuration
+ */
+struct mspi_cfg {
+	/** @brief mspi channel number */
+	uint8_t                 ui8MSPIChannel;
+	/** @brief Configure operaton mode */
+	enum mspi_op_mode       eOPMode;
+	/** @brief Configure duplex mode */
+	enum mspi_duplex        eDuplex;
+	/** @brief DQS support flag */
+	bool                    bDQS;
+	/** @brief Software managed multi-slave enable */
+	bool                    bSwMultiSlave;
+	/** @brief GPIO chip-select line (optional) */
+	struct gpio_dt_spec     *pCE;
+	/** @brief GPIO chip-select line numbers (optional) */
+	uint32_t                ui32CeGpioNum;
+	/** @brief Slave number from 0 to host controller slave limit. */
+	uint32_t                ui32SlaveNum;
+	/** @brief Maximum supported frequency in MHz */
+	uint32_t                ui32MaxFreq;
+	/** @brief Whether to re-initialize controller */
+	bool                    bReinit;
```

Comment:
As per zephyr coding guidelines, snake_case is required.
I don't think there are explicit requirements about the fact of including the variable type in the name, but I haven't seen this elsewhere in zephyr, so I'd suggest to drop.

Reply:
Do you happen to have link to the zephyr  doc saying snake case must be used? I couldn’t find it anywhere and it was why I kept it till now.

Reply:
I guess the assumption is that this is covered by Linux coding guidelines mentioned in https://docs.zephyrproject.org/latest/contribute/guidelines.html#coding-style